File: tests/python/aws-bedrock/test_openllmetry.py
"""Conformance test: OpenLLMetry (Traceloop) Bedrock instrumentation.

Exercises: converse
against a mock Bedrock server, with the Traceloop Bedrock instrumentation.
"""

from common import run, run_converse


def instrument():
    from opentelemetry.instrumentation.bedrock import BedrockInstrumentor
    BedrockInstrumentor().instrument()


if __name__ == "__main__":
    run(
        "OpenLLMetry: AWS Bedrock Conformance Test",
        instrument,
        [run_converse],
    )

File: tests/js/aws-bedrock/test_openllmetry.ts
/**
 * Conformance test: OpenLLMetry (Traceloop) JS AWS Bedrock instrumentation.
 */

import { BedrockInstrumentation } from "@traceloop/instrumentation-bedrock";
import { run } from "./common";

function instrument(bedrockModule: any) {
  const instrumentation = new BedrockInstrumentation();
  instrumentation.manuallyInstrument(bedrockModule);
}

run("OpenLLMetry JS: AWS Bedrock Conformance Test", instrument).catch((e) => { console.error(e); process.exit(1); });
